### DISCUSSION TITLE
KERNEL_MODULE_AUTOLOAD not working

### DISCUSSION THREAD

**Discussion by danny1209** [OE4T/tegra-demo-distro#207] [Q&A]
Hi all, 

I wrote a test driver module without any dependance on hardware device, it just print some text after it´s loaded.
I added these two lines in conf/local.conf
# only for the test_driver
CORE_IMAGE_EXTRA_INSTALL += "nian"  #this is my recipe
KERNEL_MODULE_AUTOLOAD += "test-driver" #this is the test driver module

I find my module "test-driver" in .../modules/extra/
but the module is not automatically loaded... do anybody know the reason?
Thanks a lot!

**Comment by madisongh** [comment]
It could be the setting is getting overridden.  If you do a `bitbake -e | grep '^KERNEL_MODULE_AUTOLOAD='`, do you see your `test-driver` mentioned, or just `nvgpu`?  (We set that variable to that value in some of the machine configurations.)

If it's just `nvgpu` being shown, try using `KERNEL_MODULE_AUTOLOAD_append = " test-driver"` and check again to make sure that both drivers are listed.

**Reply by danny1209** [reply]
Hi @madisongh 
yes, it just shows **nvgpu**. but... 
ERROR: Variable KERNEL_MODULE_AUTOLOAD_append file: /home/nian/tegra-demo-distro/build/conf/local.conf line: 31 contains an operation using the old override syntax. Please convert this layer/metadata before attempting to use with a newer bitbake.

**Reply by madisongh** [reply]
Yes, with newer branches (honister and later), you would use `:append` instead of `_append`.
